public/app/features/library-panels/state/api.ts

```diff
@@ -65,6 +65,12 @@ export async function getLibraryPanel(uid: string, isHandled = false): Promise<L
     panels: [result.model],
   });
   const { scopedVars, ...model } = dash.panels[0].getSaveModel(); // migrated panel
+
+  //These properties should not exist on LibraryPanel.model which is of type Omit<Panel, 'gridPos' | 'id' | 'libraryPanel'>
+  delete model.gridPos;
+  delete model.id;
+  delete model.libraryPanel;
+
```

Comment:
ok, this looks funky. So this will technically change the old implementation results as well, right?

Reply:
Yes.

This whole function is funky. We're creating a dummy dashboard, and in that dashboard we're creating a normal panel from the model of the library panel. We're then getting the panel from the dashboard as any. The problem then is that it's not a library panel model, it's the normal panel model. The normal panel model has these properties that should not exist on the library panel model.

This function returns `LibraryElementDTO` which is the same as `LibraryPanel`. So we're not going to be able to access those deleted properties anyway (unless we bypass the typechecking). But when we're serializing it to JSON, those properties are still there. So `libraryPanel.model` will have both `gridPos` and `libraryPanel` properties, where the `libraryPanel` property is a `LibraryPanelRef`.

Reply:
If we want to play it real safe and keep this function as it is, I think we would have to convert the library panel model to some intermediate step, and delete those properties before serializing it.

Reply:
ok, i understand this now. I don't think this should be a problem, I mean, deleting those properties is making sure the model is compatible with the underlying type, so all good IMO.